File: backend/model/manual_ticket.go
package model

import "github.com/google/uuid"

type ManualTicket struct {
	Model
	MealOption string    `json:"option"`
	FormalID   uuid.UUID `json:"formalId"`
	Formal     *Formal   `json:"-" gorm:"constraint:OnDelete:CASCADE;"`
	// TODO: should this be an enum/custom type?
	// One of: "complimentary", "ents", "standard", "guest"
	Type          string `json:"type"`
	Name          string `json:"name"`
	Justification string `json:"justification"`
	Email         string `json:"email"`
}

File: frontend/src/model/ManualTicket.ts
export type ManualTicketType = "complimentary" | "ents" | "standard" | "guest";

export interface ManualTicket {
    id: string;
    name: string;
    justification: string;
    option: string;
    formalId: string;
    type: ManualTicketType;
    email: string;
}